tools/webpack/vendors.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+const CopyWebpackPlugin = require( 'copy-webpack-plugin' );
+const { join } = require( 'path' );
+
+/**
+ * WordPress dependencies
+ */
+const DependencyExtractionWebpackPlugin = require( '@wordpress/dependency-extraction-webpack-plugin' );
+
+const copiedVendors = {
+	'inert-polyfill': [
+		'wicg-inert/dist/inert.js',
+		'wicg-inert/dist/inert.min.js',
+	],
+};
+const importedVendors = {
+	react: { import: 'react', global: 'React' },
+	'react-dom': { import: 'react-dom', global: 'ReactDOM' },
+	'react-jsx-runtime': {
+		import: 'react/jsx-runtime',
+		global: 'ReactJSXRuntime',
+	},
+};
+
+module.exports = [
+	...Object.entries( importedVendors ).flatMap( ( [ name, config ] ) => {
+		return [ 'production', 'development' ].map( ( mode ) => {
```

Comment:
@fullofcaffeine introduced with https://github.com/WordPress/gutenberg/pull/31732 `@wordpress/readable-js-assets-webpack-plugin` to better handle the production vs development version of scripts. I'm wondering if that could be integrated here as well now that Gutenberg will have to bundle React in a special way instead of using the files they distribute.

Reply:
Do you know if it actually changes the "mode" to make the generation? Because we do need more than just minified scripts, we need the mode to change.

Reply:
@jsnajdr or @WunderBart, could also have the required knowledge to confirm it could get applied here, too. I don't fully understand how it works.

Reply:
IIRC, we generate both production (by default) and development files (by unminifying the minified files, see [here](https://github.com/WordPress/gutenberg/blob/e5f24e2cbd2603e4da3b240e4c779663204aa822/packages/readable-js-assets-webpack-plugin/index.js#L7)), and then default to `production`, relying on setting the `SCRIPT_DEBUG` constant to `true` to determine which one to enqueue.

Reply:
The `readable-js-assets` plugin doesn't change the compilation mode, it's still `production`. All it does is that it captures the non-minified version of the bundle, before Terser runs, and saves it as `.js` file. Then you have two alternatives for each asset, `.js` and `.min.js`.

React has two versions that are loaded depending on `process.env.NODE_ENV`, where the `production` one is already minified. `readable-js-assets` won't switch you to the `development` one.

Reply:
By the way, this `vendors` webpack config is a consequence of abandoning the UMD builds, do I understand that correctly? Executing the UMD version in a `script` tag would automatically create the `React` global and populate it. But the CJS version doesn't do that, it only assigns stuff to `exports`. That's why we newly execute the bundler on these scripts, to produce an output that assings the CJS exports to a global.

Reply:
@jsnajdr Yes (for the umd builds)

So based on the feedback above, it seems we don't want to use `readable-js-assets` here and using both modes is better.